Enterprise-CMCS/eAPD
# 4621 Allow deleting 1st Activity for MMIS

Resolves #4621 

### Description
This change enables deleting for the first activity of an MMIS APD on the Activities Dashboard.

### Chromatic Link
https://www.chromatic.com/build?appId=61d5b948cf6f17003a12bf77&number=1366

### Significant changes or possible side effects
- Tweaked the logic for displaying the delete button and moved it into a new function `showDeleteButton`
- `Activity.test.js` changes
  - removed enzyme and replaced with react testing library
  - removed snapshot tests
  - added extra testing (see chart below for full list)

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| APD with multiple activities | activity index is given | displays activity data based on the activity index | jest |
| HITECH APD | funding source is given | displays funding source in the heading | jest |
| non-HITECH APD | funding source is given | does NOT display funding source in the heading | jest |
| An APD activity | no activity name | displays "Untitled" in the heading | jest |
| An APD activity | no funding source | displays the heading without a funding source | jest |
| An APD activity | no activity name & no funding source |  displays "Untitled" & without a funding source in the heading  | jest |
| An APD activity | rendering | displays the edit link | jest |
| An APD activity | rendering | displays the delete button | jest |
| HITECH APD | activity index is 0 | does NOT display the delete button | jest |
| Activities nav items | getting the list of activities | activities dashboard shows as the first item | jest |
| Activities nav items | a list of activities is present | each activity shows in list | jest |
| Activities nav items | there are no activities | activities dashboard is the only item in the list | jest |
| Activities nav items | specific activity's url is passed in | the activity's selected key is set to true | jest |
| Activities nav items | activities dashboard url is passed in | the activity dashboard's selected key is set to true | jest |

### Steps to manually verify this change

1. Open an MMIS APD with at least one activity
2. Navigate to the Activities Dashboard (Activities > Activities Dashboard)
3. Verify that the Delete button exists for every activity on the list (including the first!)

_Regression check_
1. Open a HITECH APD with at least one activity
2. Navigate to the Activities Dashboard (Activities > Activities Dashboard)
3. Verify that the Delete button does NOT exist for the first activity on the list (but exists for all others)

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when QA has


### This pull request can be merged when
